pkg/cloud/api/diff.go

```diff
@@ -124,18 +136,22 @@ func (d *differ[T]) do(p Path, av, bv reflect.Value) error {
 		for i := 0; i < av.NumField(); i++ {
 			afv := av.Field(i)
 			aft := av.Type().Field(i)
-			bfv := bv.Field(i)
-			fp := p.Field(aft.Name)
 
 			if aft.Name == "NullFields" || aft.Name == "ForceSendFields" {
 				continue
 			}
 
+			fp := p.Field(aft.Name)
 			switch d.traits.fieldType(fp) {
 			case FieldTypeOutputOnly, FieldTypeSystem:
 				continue
 			}
 
+			bfv := bv.FieldByName(aft.Name)
+			if !bfv.IsValid() {
```

Comment:
we are iterating here only for A struct `for i := 0; i < av.NumField(); i++ {` and we check if fields in A are present in B (it might be in different order but we need the same field name and type). We don't do the same thing the other way. There is a comment for diff function `the behavior of this is not symmetric -- diff(A,B) != diff(B,A).` 
I thought that this is the behavior we want.

Reply:
This check is used to detect the `subset of` and I added test cases for schema validation to check if structs are subset of GA in file pkg/cloud/api/resource_test.go.
Do you want me to add specific test for `diffBetweenTypes`?